### ISSUE TITLE
feature request: react-native target

### ISSUE THREAD

**Issue by punksta** [javascript-obfuscator/javascript-obfuscator#286]
In react-native `require` can't have evaluation for argument, only static strings.  I can use `// javascript-obfuscator:disable` for any `require`, but it's a lot of manual work for large project.

## Expected Behavior

I'd like to keep static strings in require or imports without adding "// javascript-obfuscator:disable"
## Current Behavior

it obfuscate the require argument moving it to global key-value object.
 
## Steps to Reproduce (for bugs)

1. npm install -g create-react-native-app
2. create-react-native-app testapp
3. create file without jsx
4. add some imports and some require
5. try to obfuscate the file and build app using `react-native run-android` or `react-native run ios`

## Your Environment
```
"dependencies": {
    "react": "16.0.0",
    "react-i18next": "^4.8.0",
    "react-native": "0.50.4"
},
"devDependencies": {
    "javascript-obfuscator": "^0.17.0",
    "babel-core": "^6.26.0",
    "babel-eslint": "^7.2.3",
    "babel-jest": "21.2.0",
    "babel-loader": "^7.0.0",
    "babel-plugin-transform-decorators-legacy": "^1.3.4",
    "babel-plugin-transform-react-remove-prop-types": "^0.4.10",
    "babel-plugin-transform-remove-console": "^6.8.5",
    "babel-preset-react-native": "4.0.0",
    "flow-bin": "^0.56.0",
    "fs-extra": "^3.0.1",
    "zip-folder": "^1.0.0"
  },
```
I use next script for obfuscation 
```
const babel = require('babel-core');
const fs = require ("fs");
const JavaScriptObfuscator = require('javascript-obfuscator');
const {promisify} = require('util');

const babelOpts = {
	ast: false,
	compact: false,
	comments: true
};

onst obsuscateOptions = {
	compact: true,
	controlFlowFlattening: false,
	controlFlowFlatteningThreshold: 1,
	deadCodeInjection: true,
	deadCodeInjectionThreshold: 1,
	debugProtection: false,
	debugProtectionInterval: false,
	disableConsoleOutput: true,
	// identifierNamesGenerator: 'mangled',
	log: false,
	renameGlobals: false,
	rotateStringArray: false,
	selfDefending: false,
	stringArray: false,
	stringArrayEncoding: false,
	target: 'browser-no-eval',
	// stringArrayThreshold: 1,
	transformObjectKeys: false,
	unicodeEscapeSequence: false
}

const transformFileAsync = promisify(babel.transformFile);
const writeFileAsync = promisify(fs.writeFileSync);

const transformFile = (filePath, babelOpts, obsuscateOptions) => {
	return transformFileAsync(filePath, babelOpts)
		.then(r => JavaScriptObfuscator.obfuscate(
			r.code,
			obsuscateOptions
		))
		.then(r => r.getObfuscatedCode())
}

const encodeFiles = (fileList) => {
	return Promise.all(fileList.map(filePath => {
		return transformFile(filePath, babelOpts, obsuscateOptions)
			.then(code => ({
				code ,
				filePath
			}))
			.then(({code, filePath}) => writeFileAsync(filePath, code))
	}))
}

const paths = ['src/example1']

encodeFiles(paths)
	.catch(e => {
		console.error(e)
	});
```

## Minimal working example that will help to reproduce issue
obfuscate this file and import it in react-native app

```
import React, { Component} from "react"
const {View} = require("react-native")

export default () => {console.log("default module")}
```

**Comment by sanex3339** [comment]
Interesting. I'll think about a way, how to handle this.

**Comment by punksta** [comment]
@sanex3339 react-native has it's own bundle tool, https://github.com/facebook/metro/
and issues about require, maybe it will help
https://github.com/facebook/metro/issues/52
https://github.com/facebook/react-native/issues/6391

**Comment by sanex3339** [comment]
Looks like i should add option like `reservedStrings` with array of strings or regexps. But it can cause a obfuscation performance drop, because every string should be checked.

**Comment by sanex3339** [comment]
So with this option you will can ignore obfuscation of all paths with regexp and manually add strings with node package names

**Comment by punksta** [comment]
maybe better to have list of js expressions types?
```
keep: ['import', 'require', 'className', 'export']
```

**Comment by sanex3339** [comment]
No. This is different subsystems. `require` is the `IdentifierNode` and you can ignore renaming of this identifier. String inside require expression - `LiteralNode` and obfuscator has `LiteralTransformer` to handle literals and extract them to the array of strings. So, best way not to connect this fix with react-native, but make a common way to exclude literals from transformation, and `reservedStrings` option - best way. And anyone can use this option in a various way.

**Comment by sanex3339** [comment]
PR:
https://github.com/javascript-obfuscator/javascript-obfuscator/pull/295

**Comment by sanex3339** [comment]
`reservedStrings` option in `master` branch. Untill `0.18.0` release, you can use version from `master` branch.

**Comment by punksta** [comment]
@sanex3339  awesome! I will test it in Monday.

**Comment by sanex3339** [comment]
`javascript-obfuscator@0.18.0` is out!

**Comment by sytler** [comment]
any updates? Im trying to use https://github.com/javascript-obfuscator/react-native-obfuscating-transformer but there is outdated version, is there a way to use it with newer 0.18.1 lib version?

**Comment by sanex3339** [comment]
This issue was fixed by adding `reservedStrings` option in `0.18.0`. (But this release was coming with some problems that will fixed soon). For any thirdparty libraries like `react-native-obfuscating-transformer` - create issue in repository of this library.

**Comment by whoami-shubham** [comment]
Hi, @sanex3339  is this library can be used to obfuscate react-native code while building apk ? haven't found any doc/article about this library with react-native and just saw in readme faq, jsx is not supported.
